docs: added docs for command (#915)
harness/canary: apps/portal/src/content/docs/components/command.mdx

---
title: Command
description: Command component documentation
---

> The `Command` component provides a flexible UI element for displaying a command palette. It is composed of several subcomponents such as `CommandInput`, `CommandList`, `CommandItem`, and `CommandShortcut` to offer a structured and customizable interface.

Reviewer: ```suggestion
The `Command` component provides a flexible UI element for displaying a command palette. It is composed of several subcomponents such as `Command.Input`, `Command.List`, `Command.Item`, and `Command.Shortcut` to offer a structured and customizable interface.
```

import { DocsPage } from "@/components/docs-page";

> import { Command } from "@harnessio/ui/components";
> import { Button } from "@harnessio/ui/components";

Reviewer: You can drop these unless you use them directly in the MDX
```suggestion
```

<DocsPage.ComponentExample
  client:only

>   code={`
>   () => {
>   const [isOpen, setIsOpen] = React.useState(false);
> 
> return (<>
> 
> <Command.Dialog open={isOpen} onOpenChange={setIsOpen}>
>   <Command.Input placeholder="Type a command or search..." />
>   <Command.List>
>     <Command.Empty>No results found.</Command.Empty>
> 
>     <Command.Group heading="Suggestions">
>       <Command.Item>Item 1</Command.Item>
>       <Command.Item>Item 2</Command.Item>
>       <Command.Separator />
>       <Command.Item>Item 4</Command.Item>
>       <Command.Item>Item 3</Command.Item>
>     </Command.Group>
> 
>     <Command.Group heading="Commands">
>       <Command.Item>
>         Command 1
>         <Command.Shortcut>⌘K</Command.Shortcut>
>       </Command.Item>
>       <Command.Item>
>         Command 2
>         <Command.Shortcut>⌘L</Command.Shortcut>
>       </Command.Item>
>     </Command.Group>
> 
>   </Command.List>
> </Command.Dialog>
> <Button onClick={() => setIsOpen(true)}>Open Command Dialog</Button>
> 
>   </>)
>   }`}

Reviewer: Indentation is a wee bit off.
```suggestion
  code={`() => {
  const [isOpen, setIsOpen] = React.useState(false);

  return (
    <>
      <Command.Dialog open={isOpen} onOpenChange={setIsOpen}>
        <Command.Input placeholder="Type a command or search..." />
        <Command.List>
          <Command.Empty>No results found.</Command.Empty>

          <Command.Group heading="Suggestions">
            <Command.Item>Item 1</Command.Item>
            <Command.Item>Item 2</Command.Item>
            <Command.Separator />
            <Command.Item>Item 4</Command.Item>
            <Command.Item>Item 3</Command.Item>
          </Command.Group>

          <Command.Group heading="Commands">
            <Command.Item>
              Command 1
              <Command.Shortcut>⌘K</Command.Shortcut>
            </Command.Item>
            <Command.Item>
              Command 2
              <Command.Shortcut>⌘L</Command.Shortcut>
            </Command.Item>
          </Command.Group>

        </Command.List>
      </Command.Dialog>

      <Button onClick={() => setIsOpen(true)}>Open Command Dialog</Button>
    </>
  )
}`}
```

      <Command.Item></Command.Item>
    </Command.Group>
    <Command.Separator />
    <Command.Group heading="Commands">
      <Command.Item>
        <Command.Shortcut>⌘K</Command.Shortcut>
      </Command.Item>
      <Command.Item>
        <Command.Shortcut>⌘L</Command.Shortcut>
      </Command.Item>
    </Command.Group>
  </Command.List>
</Command.Dialog>
```

## API Reference

### Root

The `Root` component serves as the main container for all command elements.

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description:
        "You can pass in your `Command.Input`, `Command.List`, and other subcomponents as children",
      required: true,
      value: "ReactNode",
    },
    {
      name: "className",
      description: "Additional class names to apply to the root element",
      required: false,
      value: "string",
    },
    {
      name: "label",
      description: "Accessible label for this command menu. Not shown visibly.",
      required: false,
      value: "string",
    },
    {
      name: "shouldFilter",
      description:
        "Optionally set to `false` to turn off the automatic filtering and sorting.",
      required: false,

>       value: "boolean",

Reviewer: Should probably add the `defaultValue` for things that are optional booleans
```suggestion
      value: "boolean",
      defaultValue: "true"
```

    },
    {
      name: "filter",
      description:
        "Custom filter function for whether each command menu item should match the given search query.",
      required: false,
      value: "(value: string, search: string, keywords?: string[]) => number",
    },
    {
      name: "defaultValue",
      description: "Optional default item value when it is initially rendered.",
      required: false,
      value: "string",
    },
    {
      name: "value",
      description:
        "Optional controlled state of the selected command menu item.",
      required: false,
      value: "string",
    },
    {
      name: "onValueChange",
      description:
        "Event handler called when the selected item of the menu changes.",
      required: false,
      value: "(value: string) => void",
    },
    {
      name: "loop",
      description:
        "Optionally set to `true` to turn on looping around when using the arrow keys.",
      required: false,

>       value: "boolean",

Reviewer: Same here
```suggestion
      value: "boolean",
      defaultValue: "false"
```

    },
    {
      name: "disablePointerSelection",
      description:
        "Optionally set to `true` to disable selection via pointer events.",
      required: false,

>       value: "boolean",

Reviewer: Same here
```suggestion
      value: "boolean",
      defaultValue: "false"
```

    },
    {
      name: "vimBindings",
      description:
        "Set to `false` to disable ctrl+n/j/p/k shortcuts. Defaults to `true`.",
      required: false,

>       value: "boolean",

Reviewer: Same here
```suggestion
      value: "boolean",
      defaultValue: "true"
```
